frontend_tests/zjsunit/namespace.js

```diff
@@ -15,16 +16,22 @@ exports.set_global = function (name, val) {
     return val;
 };
 
-exports.zrequire = function (name, fn) {
+exports.zrequire = function (name, fn, stub = null) {
     if (fn === undefined) {
         fn = '../../static/js/' + name;
     } else if (/^generated\/|^js\/|^shared\/|^third\//.test(fn)) {
         // FIXME: Stealing part of the NPM namespace is confusing.
         fn = '../../static/' + fn;
     }
     delete require.cache[require.resolve(fn)];
+    var obj;
+    if (!stub) {
+        obj = require(fn);
+    } else {
+        obj = rewiremock.proxy(() => require(fn), stub);
```

Comment:
I think we should have the tests use `rewiremock` directly instead of building an additional API into `zrequire`. The only value added by `zrequire` is

1. the magical `../../static/js/` path heuristic, which is silly;
2. the cache cleaning, which `rewiremock` already does.

Using `rewiremock` directly gives will give us the flexibility to use new parts of its API without building up lots of wrapper goo. In the long term I’d like to get rid of `zrequire` entirely.